apps/arweave/src/ar_tx.erl

```diff
@@ -559,32 +559,54 @@ get_tx_fee(Args) ->
 	{DataSize, Rate, PricePerGiBMinute, KryderPlusRateMultiplier, Addr, Timestamp, Accounts,
 			Height} = Args,
 	Fork_2_6 = ar_fork:height_2_6(),
+	Fork_2_6_8 = ar_fork:height_2_6_8(),
 	Args2 = {DataSize, Rate, Height, Accounts, Addr, Timestamp},
-	case Height >= Fork_2_6 of
-		false ->
-			get_tx_fee_pre_fork_2_6(Args2);
-		true ->
-			Args3 = {DataSize, PricePerGiBMinute, KryderPlusRateMultiplier, Addr, Accounts,
+	Args3 = {DataSize, PricePerGiBMinute, KryderPlusRateMultiplier, Addr, Accounts,
 					Height},
-			case ar_pricing:is_v2_pricing_height(Height) of
-				true ->
-					get_tx_fee2(Args3);
-				false ->
-					TransitionStart = Fork_2_6 + ?PRICE_2_6_TRANSITION_START,
-					TransitionEnd = TransitionStart + ?PRICE_2_6_TRANSITION_BLOCKS,
-					case Height >= TransitionStart of
-						true ->
-							Fee1 = get_tx_fee_pre_fork_2_6(Args2),
-							Fee2 = get_tx_fee2(Args3),
-							Interval1 = Height - TransitionStart + 1,
-							Interval2 = TransitionEnd - (Height + 1),
-							(Fee1 * Interval2 + Fee2 * Interval1) div (Interval1 + Interval2);
-						false ->
-							get_tx_fee_pre_fork_2_6(Args2)
-					end
-			end
+
+	V2PricingHeight = Fork_2_6_8 + (?PRICE_2_6_8_TRANSITION_START)
+					+ (?PRICE_2_6_8_TRANSITION_BLOCKS),
+
+	TransitionStart_2_6 = Fork_2_6 + ?PRICE_2_6_TRANSITION_START,
+	TransitionEnd_2_6 = TransitionStart_2_6 + ?PRICE_2_6_TRANSITION_BLOCKS,
+	TransitionStart_2_6_8 = Fork_2_6_8 + ?PRICE_2_6_8_TRANSITION_START,
+	TransitionEnd_2_6_8 = TransitionStart_2_6_8 + ?PRICE_2_6_8_TRANSITION_BLOCKS,
+
+	%% div the fee (in winstons) by the number of bytes in a GiB rather than the DataSize
+	%% so that we can ignore/truncate any fractions (since we don't care about fractional
+	%% winstons, but we do care about factional GiBs).
+	StaticFee = (?STATIC_2_6_8_FEE_WINSTON div ?GiB) * (DataSize + ?TX_SIZE_BASE),
+
+	case Height of
+		H when H >= V2PricingHeight ->
+			get_tx_fee2(Args3);
+		H when H >= TransitionStart_2_6_8 ->
+			%% For 2.6.8 the fee at the start of the transition period is
+			%% the 2.6 transition fee in effect when the 2.6.8 fork occured.
+			get_transition_tx_fee(
+				StaticFee, %% StartFee
+				get_tx_fee2(Args3), %% EndFee
+				TransitionStart_2_6_8, 
+				TransitionEnd_2_6_8,
+				Height);
+		H when H >= Fork_2_6_8 ->
+			StaticFee;
+		H when H >= TransitionStart_2_6 ->
+			get_transition_tx_fee(
+				get_tx_fee_pre_fork_2_6(Args2), %% StartFee
+				get_tx_fee2(Args3), %% EndFee
+				TransitionStart_2_6, 
+				TransitionEnd_2_6,
+				Height);
+		_ ->
```

Comment:
We know `H >= TransitionStart_2_6` happened so we do not need this clause.

Reply:
This last clause is for H < TransitionStart_2_6 - do we still ned this clause in order to allow validation of old blocks?

Reply:
https://github.com/ArweaveTeam/arweave/pull/422/commits/15e10af482b0e9e58e95c43981386e182e7f2fdb

Reply:
Removed that clause: https://github.com/ArweaveTeam/arweave/pull/422/commits/3d33e3fe750714dac7373eddf9ad76ae5bd8f9cc